packages/client/ts/controllers/info.ts

```diff
@@ -21,18 +21,18 @@ export default class InfoController {
         this.destroyQueue = new Queue();
     }
 
-    create(type: number, data: any[], x: number, y: number): void {
+    create(type: number, data: unknown[], x: number, y: number): void {
         switch (type) {
             case Modules.Hits.Damage:
             case Modules.Hits.Stun:
             case Modules.Hits.Critical: {
-                let damage = data.shift();
-                const isTarget: boolean = data.shift();
-                const dId = this.generateId(this.game.time, damage, x, y);
+                const damage = data.shift() as number | string;
+                const isTarget = data.shift() as boolean;
+                const dId = this.generateId(this.game.time, damage as number, x, y);
 
-                if (damage < 1 || !isInt(damage)) damage = 'MISS';
+                if (damage < 1 || !isInt(damage as number)) (damage as string) = 'MISS';
```

Comment:
Ok, I see that `data` can have "MISS" string. Are there any other strings possible? How about using "-1" as a miss? This is just a proposal, we can discuss it and create some follow-up issues.